app/javascript/src/NewService/index.jsx

```diff
@@ -0,0 +1,127 @@
+const NewServiceWrapper = (props: Props) => {
+  const [isServDescVisible, setServDescVisible] = useState(false)
+  const [isServNewVisible, setServNewVisible] = useState(true)
+  const [namespaces, setNamespaces] = useState([])
+  const [services, setServices] = useState([])
+  const [selectedNamespace, setSelectedNamespace] = useState(undefined)
+  const [servicesRequested, setServicesRequested] = useState(false)
+
+  const settingSelectedNamespace = (namespace: string) => {
+    setSelectedNamespace(namespace)
+  }
+
+  const onChangeNamespaces = (event: SyntheticEvent<HTMLSelectElement>) => {
+    setSelectedNamespace(event.currentTarget.value)
+    setServicesRequested(true)
+  }
+  // eslint-disable-next-line flowtype/no-weak-types
+  const startNamespacesFetch = async (): Promise<any> => {
+    const namespacesData = await fetchData(NAMESPACES_URL)
+    if (namespacesData) {
+      setNamespaces(namespacesData)
+      setServicesRequested(true)
+      settingSelectedNamespace(namespacesData[0])
+    }
+  }
+
+  const startServicesFetch = async (servicesUrl: string) => {
+    setServices([])
+    const servicesData = await fetchData(servicesUrl)
+    if (servicesData) {
+      setServices(servicesData)
+      setServicesRequested(false)
+    }
+  }
+
+  const handleFormsVisibility = (event: SyntheticEvent<HTMLInputElement>) => {
+    if (event.currentTarget.value === 'discover') {
+      setServDescVisible(true)
+      setServNewVisible(false)
+    } else {
+      setServDescVisible(false)
+      setServNewVisible(true)
+    }
+  }
+
+  useEffect(() => {
+    if (servicesRequested && selectedNamespace) {
+      const SERVICES_URL = `${BASE_URL}namespaces/${selectedNamespace}/services.json`
+      startServicesFetch(SERVICES_URL)
+    }
+  }, [servicesRequested, selectedNamespace])
+
+  const formsProps: {
+    source: SourceFormType,
+    serviceDiscovery: ServiceDiscoveryFormType,
+    serviceNew: ServiceNewFormType
+  } = {
+    source: {
+      isServiceDiscoveryUsable: props.isServiceDiscoveryUsable,
+      serviceDiscoveryAuthenticateUrl: props.serviceDiscoveryAuthenticateUrl,
+      onHandleFormsVisibility: handleFormsVisibility,
+      onStartNamespacesFetch: startNamespacesFetch
+    },
+    serviceDiscovery: {
+      isVisible: isServDescVisible,
+      providerAdminServiceDiscoveryServicesPath: props.providerAdminServiceDiscoveryServicesPath,
+      namespaces: namespaces,
+      services: services,
+      selectedNamespace: selectedNamespace,
+      handleChangeNamespaces: onChangeNamespaces
+    },
+    serviceNew: {
+      isVisible: isServNewVisible,
+      adminServicesPath: props.adminServicesPath
+    }
+  }
+
+  return (
+    <React.Fragment>
+      <h1>NEW API</h1>
+      <ServiceSourceForm {...formsProps.source}/>
+      <ServiceDiscoveryForm {...formsProps.serviceDiscovery}/>
+      <ServiceNewForm {...formsProps.serviceNew}/>
+    </React.Fragment>
+  )
+}
+
+export default NewServiceWrapper
```

Comment:
I think this is wrong. We have been using index files to export all components from a module. `NewServiceWrapper` should have its own file and be exported as usual.

About naming, I don't think this is a wrapper but the form itself. It should be `NewApiForm` or `NewServiceForm`, whereas the rest are the wrappers.

We should also implement conditional rendering whenever it's possible: using `display: none` is not the React way and also it is implicit. Taking a look at this render method and the state etc we should get a general idea of what this component does and how it behaves. Don't get me wrong, right now it is not _bad_ but it could be more readable if we get rid of some redundant props and return a more explicit composition.